crates/rpc/rpc/src/engine.rs

```diff
@@ -43,7 +44,7 @@ where
             RpcReceipt<Eth::NetworkTypes>,
             RpcHeader<Eth::NetworkTypes>,
         > + FullEthApiTypes,
-    EthFilter: EthFilterApiServer<RpcTransaction<Eth::NetworkTypes>>,
+    EthFilter: EngineEthFilter + Send + Sync + 'static,
```

Comment:
I think we can move this bounds to the trait def and require this for the trait itself